pmlc/dialect/pxa/transforms/dealloc_placement.cc

```diff
@@ -57,102 +56,48 @@ struct DeallocPlacementPass
 
   template <typename Callback>
   void runOnFunction(FuncOp fn, Callback onPack) {
-    // Place deallocation for AllocOp
     fn.walk([&](AllocOp alloc) {
       IVLOG(3, "alloc: " << debugString(*alloc));
-      placeDealloc(alloc.getResult(), alloc, alloc.getOperation()->getBlock(),
-                   -1, onPack);
-    });
-    // Place deallcation for the loop (scf.for) arguments
-    fn.walk([&](scf::ForOp forOp) {
-      auto args = forOp.getRegionIterArgs();
-      for (unsigned i = 0; i < args.size(); ++i) {
-        placeDealloc(cast<Value>(args[i]), &forOp.getBody()->front(),
```

Comment:
this code will generate the dealloc op, it will dealloc the scf.for iterator. and I believe that will definitely run into some error 
```
malloc_consolidate(): invalid chunk size
Process finished with exit code 134 (interrupted by signal 6: SIGABRT)
```
I can't figure out why yao add this op, maybe we should ask him about this PR.

Reply:
Let's ask yao about it tomorrow.